projects/packages/connection/src/class-users-connection-admin.php

```diff
@@ -32,7 +32,7 @@ public function __construct() {
 	 * Initialize the admin functionality if conditions are met.
 	 */
 	public function init() {
-		if ( ! is_admin() || ! current_user_can( 'manage_options' ) ) {
+		if ( ! is_admin() || ! current_user_can( 'manage_options' ) || ( defined( 'IS_WPCOM' ) || IS_WPCOM ) ) {
```

Comment:
I think you should be able to rely on `Automattic\Jetpack\Status\Host` and its `is_wpcom_simple` method here instead of the constant.